public/components/common/tag_filter_popover_content/tag_filter_popover_content.tsx

```diff
@@ -0,0 +1,232 @@
+    return (
+      <TagValueSelector
+        singleSelection={operator === TagFilterOperator.Is || operator === TagFilterOperator.IsNot}
+        value={typeof value === 'number' ? undefined : value}
+        onChange={onChange}
+      />
+    );
+  }
+  return (
+    <EuiFieldNumber
+      value={value === undefined ? '' : value}
+      onChange={(e) => {
+        const newValue = e.currentTarget.value;
+        onChange(newValue === '' ? undefined : parseFloat(newValue));
+      }}
+      fullWidth
+      placeholder="Add a value"
+    />
+  );
+};
+
+export const TagFilterPopoverContent = ({
+  onSave,
+  tagKeys,
+  onCancel,
+  tagFilter,
+}: TagFilterPopoverContentProps) => {
+  const [value, setValue] = useState<string | string[] | number | undefined>(tagFilter?.value);
+  const [selectedTagOptions, setSelectedTagOptions] = useState<
+    Array<EuiComboBoxOptionOption<TagKey>>
+  >(() => {
+    if (!tagFilter) {
+      return [];
+    }
+    return [
+      {
+        label: tagFilter.name,
+        value: {
+          name: tagFilter.name,
+          type: typeof tagFilter.value === 'number' ? 'number' : 'string',
+        },
+      },
+    ];
+  });
+  const [selectedOperatorOptions, setSelectedOperatorOptions] = useState<
+    Array<EuiComboBoxOptionOption<{}>>
+  >(() => {
+    if (!tagFilter) {
+      return [];
+    }
+    return [
+      {
+        label: tagFilter.operator,
+      },
+    ];
+  });
+  const selectedTag = selectedTagOptions[0]?.value;
+  const selectedTagType = selectedTag?.type;
+
+  const tagOptions = useMemo(() => tagKeys.map((item) => ({ label: item.name, value: item })), [
+    tagKeys,
+  ]);
+
+  const operatorsOptions = useMemo(() => {
+    if (!selectedTagType) {
+      return [];
+    }
+    return [
+      TagFilterOperator.Is,
+      TagFilterOperator.IsNot,
+      ...(selectedTagType === 'string'
+        ? [TagFilterOperator.IsOneOf, TagFilterOperator.IsNotOneOf]
+        : []),
+      ...(selectedTagType === 'number'
+        ? [TagFilterOperator.IsGreaterThan, TagFilterOperator.IsLessThan]
+        : []),
+    ].map((label) => ({
+      label,
+    }));
+  }, [selectedTagType]);
+  const operator = selectedOperatorOptions[0]?.label as TagFilterOperator;
+
+  const tagKeyOptionRenderer = useCallback(
+    (option: EuiComboBoxOptionOption<TagKey>, _searchValue: string, contentClassName: string) => {
+      return (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+          <EuiToken
+            style={{ marginRight: 2 }}
+            iconType={option.value?.type === 'number' ? 'tokenNumber' : 'tokenString'}
+          />
+          <span className={contentClassName}>{option.label}</span>
+          <EuiText style={{ marginLeft: 'auto' }} color="subdued" size="s">
+            {option.value?.type}
+          </EuiText>
+        </div>
+      );
+    },
+    []
+  );
+
+  const handleSave = useCallback(() => {
+    if (!selectedTag || !operator || !value) {
```

Comment:
Shall disable `Save` button if `!selectedTag || !operator || !value`?

Reply:
This assert is more for type safe. In general, the selectedTag and operator should not be Falsy if value is Truly.